src/cart/crt_iv.c

```diff
@@ -24,6 +24,8 @@
 #define IV_DBG(key, msg, ...) \
 	D_DEBUG(DB_TRACE, "[key=%p] " msg, (key)->iov_buf, ##__VA_ARGS__)
 
+/* Clone DAOS_BULK_LIMIT for the moment XXX */
+#define CRT_IV_BULK_LIMIT (20480 - 1024)
```

Comment:
I agree that this needs more thought/justification.  Does this not work above the bulk limit or does it with a performance change?  Where do the two numbers above actually come from, should we be looking to tune this at all?

Reply:
Sorry, I do not see "eager limit size" in our code, where is it? @frostedcmos   

I thought about using DAOS_BULK_LIMIT in crt_iv.c, though it seems not easy to export any daos definition/api to cart for now?  any thought?

I think DAOS_BULK_LIMIT is provider independent, so I just re-use this value, which makes more sense to me. Though I do not know how this number was gotten(PR-6599), with what provider.

Reply:
The main number is this one:
```
(
980 /** Size of (un)expected Mercury buffers */
 981 #define DAOS_RPC_SIZE  (20480) /* 20KiB */
)
```

This threshold can be adjusted up to some maximum supported by each provider. Experimentally johannn previously found that 20k is a good size for performance between inline and rdma on verbs and tcp i think. 

If cart sends inline data bigger than the limit that it specified to the mercury then internally mercury will perform a bulk transfer of that data, defeating the whole purpose of this PR:)

That limit today can be set using crt_init_options_t::cio_max_expected/unexpected_size.

It can be also overidden using DAOS_RPC_SIZE_LIMIT envariable

see cart/crt_init.c:
```
 665         /** configure cart for maximum bulk threshold */
666         d_getenv_int("DAOS_RPC_SIZE_LIMIT", &limit);
667
668         daos_crt_init_opt.cio_use_expected_size = 1;
669         daos_crt_init_opt.cio_max_expected_size = limit ? limit : DAOS_RPC_SIZE;
670         daos_crt_init_opt.cio_use_unexpected_size = 1;
671         daos_crt_init_opt.cio_max_unexpected_size = limit ? limit : DAOS_RPC_SIZE;
```

Reply:
> I thought about using DAOS_BULK_LIMIT in crt_iv.c, though it seems not easy to export any daos definition/api to cart for now? any thought?

Seems the proper way here would be to use the actual limit that was set at runtime which is today stored in:
 215         prov_data->cpg_max_exp_size = max_expect_size;
 216         prov_data->cpg_max_unexp_size = max_unexpect_size;

(We should also probably for now enforce those 2 values to be always the same)

You will need to add new accessor function similar to:
` 
src/cart/crt_hg.c
588 crt_provider_get_max_ctx_num(bool primary, int provider)
`

Reply:
ah, ok, thanks.